src/Provider/OGCWebServiceHelper.js

```diff
@@ -23,64 +17,49 @@ const tileCoord = new Extent('WMTS:WGS84G', 0, 0, 0);
 
 export default {
     getColorTextureByUrl(url, networkOptions) {
-        if (cache.has(url)) {
-            return Promise.resolve(cache.get(url));
+        if (Cache.has(url)) {
+            return Promise.resolve(Cache.get(url));
         }
 
-        const promise = (pending.has(url)) ?
-            pending.get(url) :
-            Fetcher.texture(url, networkOptions);
-
-        pending.set(url, promise);
-
-        return promise.then((texture) => {
+        const promise = Fetcher.texture(url, networkOptions).then((texture) => {
             texture.generateMipmaps = false;
             texture.magFilter = THREE.LinearFilter;
             texture.minFilter = THREE.LinearFilter;
             texture.anisotropy = 16;
 
-            if (!cache.has(url)) {
-                cache.set(url, texture);
-            }
-
-            pending.delete(url);
+            Cache.set(url, texture, Cache.POLICIES.TEXTURE);
```

Comment:
I think it's not needed anymore because you already saved this request in the cache at line 34.

Reply:
Maybe you could adapt the Cache API to get `Cache.set` return the entry. Then you could use this pattern:
```js
getColorTextureByUrl(url, networkOptions) {
    return Cache.get(url) || Cache.set(url, Fetcher.texture(url, networkOptions).then((texture) => {
             texture.generateMipmaps = false;
             texture.magFilter = THREE.LinearFilter;
             texture.minFilter = THREE.LinearFilter;
             texture.anisotropy = 16;
    }));
}
```
WDYT?

Reply:
Oh I didn't know I could do something like this. I'll try it !